caffe2/CMakeLists.txt

```diff
@@ -1314,7 +1314,12 @@ if (BUILD_PYTHON)
   target_link_libraries(
       caffe2_pybind11_state torch_library)
   if (WIN32)
-    target_link_libraries(caffe2_pybind11_state ${PYTHON_LIBRARIES})
+    if ((GENERATOR_IS_MULTI_CONFIG AND "Debug" IN_LIST CMAKE_CONFIGURATION_TYPES)
+      OR ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+      target_link_libraries(caffe2_pybind11_state ${PYTHON_DEBUG_LIBRARIES})
```

Comment:
I guess a better solution for multiple-build-type generators is to use generator expressions.
Should be sth like `target_link_libraries(caffe2_pybind11_state  $<$<CONFIG:Debug>:${PYTHON_DEBUG_LIBRARIES}> $<$<NOT:$<CONFIG:Debug>>:${PYTHON_LIBRARIES}>)`.

Reply:
We need to change these conditions thoroughly.